substrate-node/pallets/pallet-tfgrid/src/node.rs

```diff
@@ -358,14 +358,17 @@ impl<T: Config> Pallet<T> {
         let twin_id = TwinIdByAccountID::<T>::get(account_id).ok_or(Error::<T>::TwinNotExists)?;
         let node = Nodes::<T>::get(node_id).ok_or(Error::<T>::NodeNotExists)?;
         let farm = Farms::<T>::get(node.farm_id).ok_or(Error::<T>::FarmNotExists)?;
+
+        // Make sure only the farmer that owns this node can change the power target
         ensure!(
             twin_id == farm.twin_id,
             Error::<T>::UnauthorizedToChangePowerTarget
         );
-        // Make sure only the farmer that owns this node can change the power target
+
+        // Make sure there are no active contracts on node
         ensure!(
-            node.farm_id == farm.id,
-            Error::<T>::UnauthorizedToChangePowerTarget
+            T::NodeActiveContracts::node_has_no_active_contracts(node_id),
+            Error::<T>::NodeHasActiveContracts
```

Comment:
This condition should only be checked when the farmer intends to bring their node down. Otherwise, it will prevent the farmer from waking up their node if it has been rented.

Reply:
Indeed